app/PermissionsContext.js

```diff
@@ -0,0 +1,141 @@
+import React, { createContext, useEffect, useState } from 'react';
+import { Platform } from 'react-native';
+import {
+  PERMISSIONS,
+  check,
+  checkNotifications,
+  request,
+  requestNotifications,
+} from 'react-native-permissions';
+
+import { config } from './COVIDSafePathsConfig';
+import { HCAService } from './services/HCAService';
+
+export const PermissionStatus = {
+  UNKNOWN: 0,
+  GRANTED: 1,
+  DENIED: 2,
+};
+
+const statusToEnum = status => {
+  switch (status) {
+    case 'unknown': {
+      return PermissionStatus.UNKNOWN;
+    }
+    case 'denied': {
+      return PermissionStatus.DENIED;
+    }
+    case 'blocked': {
+      return PermissionStatus.DENIED;
+    }
+    case 'granted': {
+      return PermissionStatus.GRANTED;
+    }
+    default: {
+      return PermissionStatus.UNKNOWN;
+    }
+  }
+};
+
+const PermissionsContext = createContext({
+  locationPermission: PermissionStatus.UNKNOWN,
+  requestLocationPermission: () => {},
+  notificationPermission: PermissionStatus.UNKNOWN,
+  requestNotificationPermission: () => {},
+  authSubscriptionPermission: PermissionStatus.UNKNOWN,
+  requestAuthSubscriptionPermission: () => {},
+});
+
+const PermissionsProvider = ({ children }) => {
+  const { tracingStrategy } = config;
+  const [locationPermission, setLocationPermission] = useState(
+    PermissionStatus.UNKNOWN,
+  );
+  const [notificationPermission, setNotificationPermission] = useState(
+    PermissionStatus.UNKNOWN,
+  );
+  const [authSubscriptionPermission, setAuthSubscriptionPermission] = useState(
+    PermissionStatus.UNKNOWN,
+  );
+
+  const isGPS = tracingStrategy === 'gps';
+
+  useEffect(() => {
+    if (isGPS) {
+      checkLocationPermission();
+    }
+
+    if (Platform.OS === 'ios') {
+      checkNotificationStatus();
+    }
+
+    if (__DEV__ && isGPS) {
+      checkSubsriptionStatus();
+    }
+  }, []);
+
+  const checkLocationPermission = async () => {
+    const status = await check(PERMISSIONS.IOS.LOCATION_ALWAYS);
+    setLocationPermission(statusToEnum(status));
+  };
+
+  const checkNotificationStatus = async () => {
+    const { status } = await checkNotifications();
+    setNotificationPermission(statusToEnum(status));
+  };
+
+  const checkSubsriptionStatus = async () => {
+    const hasUserSetSubscription = await HCAService.hasUserSetSubscription();
```

Comment:
When pulling out this stuff, I was a little confused at what was going on, since it defaulted to 'granted' no matter what and didn't seem to be doing much yet.  So I tried to keep feature parity, but there wasnt a clear way for me to tell if anything changed/broke.

That said, given that this is only happening on the dev builds i didnt spend too much time digging into it, so if anyone has some context here and could verify if didnt break something for someone that would be much appreciated.

Reply:
@Patrick-Erichsen should have more context here

maybe it was an oversight, or intentional? either way lets add a comment to the code if it's intentional.

For now, to unblock, you could add a `// TODO(<SAF-XX-link>): blah` and add a jira task to follow up and fix/clarify the HCA auto sub logic.

Reply:
Thanks for the context!

Given that it only exists during the onboarding flow and is not really a device permission would it make sense to remove it from the permissions context?

Reply:
Yes, it makes sense to remove it from "permissions". But I think it is helpful to know if the value is set to false or not (undefined).. i.e. when upgrading to a new version